src/browser/BLECentralPlugin.js

```diff
@@ -1,63 +1,194 @@
-    },
-    startScan: function(services, success, failure) {
-        notSupported();
-        if (failure) failure();
+
+        if (!navigator.bluetooth) {
+            failure('Bluetooth is not supported on this browser.');
+            return;
+        }
+
+        this.scanning = true;
+
+        let options = {};
+
+        if (services && services.length) {
+            options.filters = [{
+                services: services.map(getDeviceId)
+            }];
+        } else {
+            options.acceptAllDevices = true;
+        }
+
+        console.log('requestDevice', options);
+        navigator.bluetooth.requestDevice(options).then(device => {
+            if (this.scanning) {  
+                return device.gatt.connect().then(server => {
+                    this.devices.set(device.id, {
+                        device: device,
+                        server: server
+                    })
+                    success(device);
+                });              
+            }
+        }).catch(failure);
     },
     stopScan: function(success, failure) {
-        notSupported();
-        if (failure) failure();
+        this.scanning = false;
+        if (success) success();
     },
     startScanWithOptions: function(services, options, success, failure) {
-        notSupported();
-        if (failure) failure();
+        this.scanning = true;
+        navigator.bluetooth.requestDevice(options).then(device => {
+            if (this.scanning) {                
+                success(device);
+            }
+        }).catch(failure);
     },
     connect: function(device_id, connectSuccess, connectFailure) {
-        notSupported();
-        if (connectFailure) connectFailure();
-    },
-    disconnect: function(device_id, connectSuccess, connectFailure) {
-        notSupported();
-        if (connectFailure) connectFailure();
+        console.log('connect', device_id);
+        if (this.devices.has(device_id)) {
+            connectSuccess(this.devices.get(device_id).device);
+        } else {
+            navigator.bluetooth.requestDevice({filters: [{
+                services: [getDeviceId(device_id)]
+            }]}).then(device => {
+                return device.gatt.connect().then(server => {
+                    this.devices.set(device_id, {
+                        device: device,
+                        server: server
+                    })
+                    connectSuccess(device);
+                });
+            }).catch(connectFailure);
+        }
+    },
+    disconnect: function(device_id, disconnectSuccess, disconnectFailure) {
+        if (this.devices.has(device_id)) {
+            var device = this.devices.get(device_id).device;
+            this.devices.delete(device_id);
+            if (device.gatt.connected) {
+                device.gatt.disconnect();
+                disconnectSuccess(device);
+            } else {
+                disconnectFailure("Device not connected");
```

Comment:
Probably best not to report any error here if disconnect is called and the device isn't found. This will keep it aligned with the other platforms 😺